packages/flutter/lib/src/gestures/events.dart

```diff
@@ -338,6 +410,61 @@ abstract class PointerEvent extends Diagnosticable {
   String toStringFull() {
     return toString(minLevel: DiagnosticLevel.fine);
   }
+
+  /// Returns the transformation of `position` into the coordinate system
+  /// described by `transform`.
+  ///
+  /// The z-value of `position` is assumed to be 0.0. If `transform` is null,
+  /// `position` is returned as-is.
+  static Offset transformPosition(Matrix4 transform, Offset position) {
+    if (transform == null) {
+      return position;
+    }
+    final Vector3 position3 = Vector3(position.dx, position.dy, 0.0);
+    final Vector3 transformed3 = transform.perspectiveTransform(position3);
+    return Offset(transformed3.x, transformed3.y);
+  }
```

Comment:
It's a pity we can't easily put this on Offset itself (but we can't because dart:ui can't import Matrix4, so it'd have to expose an API that took a Float64List instead.)